File: abc/abc311/a.java
// https://atcoder.jp/contests/abc311/submissions/43891044

import java.util.Scanner;

class Main {
  public static void main(String args[]) {
    Scanner stdIn = new Scanner(System.in);

    int n = stdIn.nextInt();
    String s = stdIn.next();

    boolean[] check = new boolean[3];
    for(int i = 0; i < 3; i++) check[i] = false;

    for(int i = 0; i < n; i++) {
      check[(int)s.charAt(i) - 65] = true;

      boolean flag = true;
      for(int j = 0; j < 3; j++) {
        if(!check[j]) {
          flag = false;
        }
      }

      if(flag) {
        System.out.println(i + 1);
        System.exit(0);
      }
    }
  }
}

File: abc/abc311/a.py
# https://atcoder.jp/contests/abc311/submissions/43890907

n = int(input())
s = input()

set = set()
for i in range(n):
  set.add(s[i])

  check = True
  for j in range(65, 68):
    c = chr(j)
    if c not in set:
      check = False

  if check:
    print(i + 1)
    exit()

